### PR TITLE
Add unit test to cicd

### PR DESCRIPTION
**Pull request by ramya-subramanyam**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

Add HIL tests to run on our CI/CD.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -70,22 +70,6 @@ jobs:
           kill $(cat server.pid)
           rm server.pid
 
-      - name: Remove latest installed platform version
```
Comment (jaenrig-ifx) [inline .github/workflows/build.yml:73]: I understand that these steps are not really necessary as they are taking care by the arduino-cli installation or?

**Review by jaenrig-ifx — COMMENTED** [review]
I see that all this is building in the labpc. And it only fails in the flashing or?
First we would need to attach some board to the HIL. I think there is none yet.

**Comment by jaenrig-ifx** [comment]
I mean, in general I would try to move asap  all these things to the arduino-core-test and pull it in here.
The more we develop in parallel de more deviations and tech debt in future integration. 

Then let´s try with the rest to get those workflows leaner, with some minimal calls to encapsulated functionality in scripts (that can also be run locally), etc.

But the first next step, I would say is to try to move all common content to the arduino-core-test, and consume that from this repo (and the xmc-for-arduino).

Anyhow, if the current hil workflow gets to work, we can just merge :)

**Comment by ramya-subramanyam** [comment]
> I see that all this is building in the labpc.... Which one? The one in Muc or in BLR? And it only fails in the flashing or?

build and tests are running only in BLR LabPC for now.

File: .github/workflows/build.yml (hunk 1 above)
Comment (ramya-subramanyam) [reply .github/workflows/build.yml:73]: Thats right! We don't need to do any copy and linking here. This approach is working very well and stable. I guess we can take this workflow also for XMC.

**Comment by ramya-subramanyam** [comment]
> I mean, in general I would try to move asap all these things to the arduino-core-test and pull it in here. The more we develop in parallel de more deviations and tech debt in future integration.
> 
> Then let´s try with the rest to get those workflows leaner, with some minimal calls to encapsulated functionality in scripts (that can also be run locally), etc.
> 
> But the first next step, I would say is to try to move all common content to the arduino-core-test, and consume that from this repo (and the xmc-for-arduino).
> 
> Anyhow, if the current hil workflow gets to work, we can just merge :)

Yes I will not work on this parallelly anymore until we have a generic approach of hil support from Olaf. 
Right now we can add tests and all tests should run.

**Review by IFX-Anusha — APPROVED** [review]
This looks good to me. But can't comment much as my involvement was limited on this topic.

**Review by jaenrig-ifx — APPROVED** [review]

**Pull request merged**